compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/HttpServiceValidator.java

```diff
@@ -79,6 +93,48 @@ public void perform(SyntaxNodeAnalysisContext syntaxNodeAnalysisContext) {
         }
     }
 
+    private static void extractServiceAnnotationAndValidate(SyntaxNodeAnalysisContext ctx,
+                                                            ServiceDeclarationNode serviceDeclarationNode) {
+        Optional<MetadataNode> metadataNodeOptional = serviceDeclarationNode.metadata();
+
+        if (metadataNodeOptional.isEmpty()) {
+            return;
+        } else {
+            NodeList<AnnotationNode> annotations = metadataNodeOptional.get().annotations();
+            for (AnnotationNode annotation : annotations) {
+                Node annotReference = annotation.annotReference();
+                String annotName = annotReference.toString();
+                Optional<MappingConstructorExpressionNode> annotValue = annotation.annotValue();
+                if (annotReference.kind() != SyntaxKind.QUALIFIED_NAME_REFERENCE) {
+                    continue;
+                }
+                String[] annotStrings = annotName.split(COLON);
+                if (SERVICE_CONFIG_ANNOTATION.equals(annotStrings[annotStrings.length - 1].trim())
+                        && (annotValue.isPresent())) {
+                    MappingConstructorExpressionNode mapping = annotValue.get();
```

Comment:
Extract code from L114 to L132 to another method called `validateServiceConfigAnnotation.